### PR TITLE
[PLATFORM-1374] Handle draft product loading permission

### PR DESCRIPTION
**Pull request by juhah**
Fix product loading so that only published products can be accessed on the product page. Trying to view a draft product now shows 404 page.

### CODE REVIEW

File: app/src/auth/utils/loginInterceptor.js [hunk 1]
```diff
@@ -99,27 +99,27 @@ export function canHandleLoadError(err) {
...
-        throw err
+        throw error
     }
 
-    if (status === 404 || ([401, 403].includes(status) && isLoggedInError(err))) {
-        const data = err.response.data || {}
-        throw new ResourceNotFoundError(data.type, data.id)
+    if (status === 404 || ([401, 403].includes(status) && (!!ignoreUnauthorized || isLoggedInError(error)))) {
+        const data = error.response.data || {}
+        throw new ResourceNotFoundError(data.type || data.resource, data.id)
```
Comment (juhah) [inline app/src/auth/utils/loginInterceptor.js:115]: Resource type is inconsistent in the backend, can be `type` or `resource`.

For example `/streams/id`:

Screen Shot 2020-06-15 at 15 39 32

`/streams/id/permissions/me`:

Screen Shot 2020-06-15 at 15 38 17

File: app/src/shared/components/NotFoundPage/index.jsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ const NotFoundPage = () => (
                         <Button
                             kind="special"
                             tag={Link}
-                            to={routes.core()}
+                            to={routes.marketplace.index()}
```
Comment (juhah) [inline app/src/shared/components/NotFoundPage/index.jsx:42]: Fix marketplace link in 404 page ([PLATFORM-1400](https://streamr.atlassian.net/browse/PLATFORM-1400))

**Comment by mondoreale** [comment]
Would you mind adding 2-3 e2e tests for it?

**Comment by juhah** [comment]
> Would you mind adding 2-3 e2e tests for it?
 
Good idea, I will try adding some Cypress tests for these

**Comment by juhah** [comment]
Added some tests in https://github.com/streamr-dev/streamr-platform/pull/1027/commits/814a8f21b99fecc0902fa2f96b720b5da8d2f771

**Pull request merged**